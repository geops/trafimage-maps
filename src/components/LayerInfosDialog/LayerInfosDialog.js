import React from 'react';
import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';
import { Trans, useTranslation } from 'react-i18next';
import Layer from 'react-spatial/layers/Layer';
import Dialog from '../Dialog';
import layerInfos from '../../layerInfos';

const propTypes = {
  selectedForInfos: PropTypes.object,
};

const defaultProps = {
  selectedForInfos: null,
};

export const NAME = 'infoDialog';

const getLegendUrl = (legendUrl, language) => {
  const src = legendUrl || '';
  return /{language}/.test(legendUrl)
    ? src.replace('{language}', language)
    : src;
};

function LayerInfosDialog(props) {
  const language = useSelector(state => state.app.language);
  const { t } = useTranslation();
  const { selectedForInfos } = props;

  if (!selectedForInfos) {
    return null;
  }

  const componentName =
    selectedForInfos instanceof Layer
      ? selectedForInfos.get('layerInfoComponent')
      : selectedForInfos.layerInfoComponent;

  let body;
  if (componentName) {
    const LayerInfoComponent = layerInfos[componentName];
    body = <LayerInfoComponent language={language} />;
  } else if (selectedForInfos instanceof Layer) {
    body = (
      <Trans
        i18nKey={selectedForInfos.get('description') || ''}
        components={[
          <img
            src={getLegendUrl(selectedForInfos.get('legendUrl'), language)}
            draggable="false"
            alt={t('Kein Bildtext')}
          />,
        ]}
      />
    );
  } else {
    body = (
      <Trans
        i18nKey={selectedForInfos.description || ''}
        components={[
          <img
            src={getLegendUrl(selectedForInfos.legendUrl, language)}
            draggable="false"
            alt={t('Kein Bildtext')}
          />,
        ]}
      />
    );
  }

  return (
    <Dialog
      isDraggable
      cancelDraggable=".tm-dialog-body"
      name={NAME}
      title={<span>{t('Informationen')}</span>}
      body={body}
      // eslint-disable-next-line react/jsx-props-no-spreading
      {...props}
    />
  );
}

LayerInfosDialog.propTypes = propTypes;
LayerInfosDialog.defaultProps = defaultProps;

export default LayerInfosDialog;
