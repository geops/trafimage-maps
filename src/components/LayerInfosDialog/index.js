export { default, NAME } from './LayerInfosDialog';
