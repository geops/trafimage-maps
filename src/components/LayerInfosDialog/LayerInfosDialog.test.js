import 'jest-canvas-mock';
import React from 'react';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Layer from 'react-spatial/layers/Layer';
import OLLayer from 'ol/layer/Vector';
import LayerInfosDialog from '.';

configure({ adapter: new Adapter() });

describe('LayerInfosDialog', () => {
  const mockStore = configureStore([thunk]);
  let store;
  let layer;

  beforeEach(() => {
    layer = new Layer({
      name: 'test',
      olLayer: new OLLayer(),
      properties: {
        description: 'description<br/>break',
      },
    });
    store = mockStore({
      app: {
        language: 'de',
        t: t => t,
      },
    });
  });

  test('should match snapshot when Layer is null', () => {
    const component = renderer.create(
      <Provider store={store}>
        <LayerInfosDialog />
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should match snapshot when Layer is defined', () => {
    const component = renderer.create(
      <Provider store={store}>
        <LayerInfosDialog selectedForInfos={layer} />
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
