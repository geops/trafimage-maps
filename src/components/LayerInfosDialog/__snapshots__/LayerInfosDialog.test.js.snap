// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayerInfosDialog should match snapshot when Layer is defined 1`] = `
<div
  className="wkp-dialog"
>
  <div
    className="tm-rnd tm-draggable react-draggable"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "boxSizing": "border-box",
        "cursor": "move",
        "display": "inline-block",
        "flexShrink": 0,
        "height": "auto",
        "left": 0,
        "maxHeight": 9007199254740991,
        "maxWidth": 9007199254740991,
        "minHeight": undefined,
        "minWidth": undefined,
        "position": "absolute",
        "top": 0,
        "touchAction": "none",
        "transform": "translate(0px,0px)",
        "userSelect": "auto",
        "width": "auto",
      }
    }
    transform={null}
  >
    <div
      className="tm-dialog"
      style={
        Object {
          "display": "block",
        }
      }
    >
      <div
        className="tm-dialog-container"
      >
        <div
          className="tm-dialog-header tm-dialog-draggable"
        >
          <div
            className="tm-header-left"
          />
          <span>
            Informationen
          </span>
          <div
            className="tm-button tm-dialog-close-bt"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex={0}
          >
            <svg
              fill="currentColor"
              focusable={false}
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </div>
        </div>
        <div
          className="tm-dialog-body"
        >
          description&lt;br/&gt;break
        </div>
      </div>
    </div>
    <span />
  </div>
</div>
`;

exports[`LayerInfosDialog should match snapshot when Layer is null 1`] = `null`;
